pandas/io/pytables.py

```diff
@@ -259,7 +261,13 @@ def to_hdf(path_or_buf, key, value, mode=None, complevel=None, complib=None,
                       complib=complib) as store:
             f(store)
     else:
-        f(path_or_buf)
+        path_or_buf = _stringify_path(path_or_buf)
```

Comment:
I think is redundant, just adding a ``path_or_buf = _stringify_path(path_or_buf)`` (before the if should allow this2 to work, rather than adding another test for ``if instance(path_or_buf, string_tyeps)`` (which is repeated).